ci-operator/step-registry/ipi/install/install/in-bastion/ipi-install-install-in-bastion-commands.sh

```diff
@@ -0,0 +1,222 @@
+    return 0
+}
+
+function run_scp_from_remote() {
+    local sshkey=$1
+    local user=$2
+    local host=$3
+    local src=$4
+    local dest=$5
+
+    options=" -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null "
+    cmd="scp -r ${options} -i \"${sshkey}\" ${user}@${host}:${src} ${dest}"
+    run_command "$cmd" || return 2
+    return 0
+}
+
+function populate_artifact_dir() {
+  set +e
+  echo "Copying log bundle..."
+  cp "${dir}"/log-bundle-*.tar.gz "${ARTIFACT_DIR}/" 2>/dev/null
+  echo "Removing REDACTED info from log..."
+  sed '
+    s/password: .*/password: REDACTED/;
+    s/X-Auth-Token.*/X-Auth-Token REDACTED/;
+    s/UserData:.*,/UserData: REDACTED,/;
+    ' "${dir}/.openshift_install.log" > "${ARTIFACT_DIR}/.openshift_install-$(date +%s).log"
+  sed -i '
+    s/password: .*/password: REDACTED/;
+    s/X-Auth-Token.*/X-Auth-Token REDACTED/;
+    s/UserData:.*,/UserData: REDACTED,/;
+    ' "${dir}/terraform.txt"
+  tar -czvf "${ARTIFACT_DIR}/terraform.tar.gz" --remove-files "${dir}/terraform.txt"
+}
+
+function write_install_status() {
+  #Save exit code for must-gather to generate junit
+  [[ -n "$ret" ]] && echo "$ret" >> "${SHARED_DIR}/install-status.txt"
+}
+
+function prepare_next_steps() {
+  write_install_status
+  set +e
+  echo "Setup phase finished, prepare env for next steps"
+  populate_artifact_dir
+
+  echo "Copying required artifacts to shared dir"
+  #Copy the auth artifacts to shared dir for the next steps
+  cp \
+      -t "${SHARED_DIR}" \
+      "${dir}/auth/kubeconfig" \
+      "${dir}/auth/kubeadmin-password" \
+      "${dir}/metadata.json"
+}
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+trap 'prepare_next_steps' EXIT TERM INT
+
+if [[ -z "$OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE" ]]; then
+  echo "OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE is an empty string, exiting"
+  exit 1
+fi
+
+if [[ -f "${SHARED_DIR}/bastion_public_address" ]]; then
+    BASTION_IP=$(<"${SHARED_DIR}/bastion_public_address")
+    BASTION_SSH_USER=$(<"${SHARED_DIR}/bastion_ssh_user")
+else
+    echo "This step is used to install cluster in bastion host, but could not find bastion_public_address in SHARED_DIR, exit..."
+    exit 1
+fi
+
+REMOTE_DIR="/tmp/"
+REMOTE_INSTALL_DIR="/tmp/installer/"
+REMOTE_ENV_FILE="/tmp/remote_env_file"
+dir=/tmp/installer
+mkdir "${dir}/"
+
+echo "Installing from release ${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE}"
+export SSH_PRIV_KEY_PATH=${CLUSTER_PROFILE_DIR}/ssh-privatekey
+
+# For disconnected or otherwise unreachable environments, we want to
+# have steps use an HTTP(S) proxy to reach the API server. This proxy
+# configuration file should export HTTP_PROXY, HTTPS_PROXY, and NO_PROXY
+# environment variables, as well as their lowercase equivalents (note
+# that libcurl doesn't recognize the uppercase variables).
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+    # shellcheck disable=SC1090
+    source "${SHARED_DIR}/proxy-conf.sh"
```

Comment:
Sounds like we do not need to source the proxy here.

Reply:
removed

Reply:
I guess we do not need to source it in this step, but need cat them into `remote_env_file`.

Reply:
@jinyunma I added a new comment here.

Reply:
get your point, I think we still need to source it, because it executes oc command to get console url in the end of this step, will add this part back.

Reply:
but seems that we don't need to add them into `remote_env_file`, because cluster can be accessed directly from bastion host whether it is disconnected or not, right?

Reply:
ah, you are right.